src/components/Filter/components/FilterMenu/components/Options/Options.tsx

```diff
@@ -0,0 +1,89 @@
+import { flatMap, head } from 'lodash';
+import React, { useCallback } from 'react';
+
+import { emptyStyle, optionsStyles } from './Options.style';
+import { FILTER_OPTIONS_MAX_HEIGHT } from 'components/Filter/constants';
+import type { FilterProps } from 'components/Filter/Filter.types';
+import type { ListSelection } from 'components/List';
+import List, { ListItem, ListItemText } from 'components/List';
+import { MAX_NON_VIRTUALIZED_ITEMS_FILTER } from 'components/List/utils';
+import { SELECT_ALL_OPTION } from 'components/Select/constants';
+
+export type Props = Pick<
+  FilterProps,
+  | 'items'
+  | 'onChange'
+  | 'defaultValue'
+  | 'selectedFilter'
+  | 'isSearchable'
+  | 'isVirtualized'
+  | 'isMulti'
+  | 'hasSelectAllOption'
+  | 'dataTestPrefixId'
+>;
+
+const Options: React.FCC<Props> = ({
+  items,
+  onChange,
+  defaultValue,
+  selectedFilter,
+  isVirtualized,
+  hasSelectAllOption,
+  isMulti,
+  dataTestPrefixId,
+}) => {
+  const isForcedVirtualized = items.length > MAX_NON_VIRTUALIZED_ITEMS_FILTER;
+
+  const height = isForcedVirtualized ? FILTER_OPTIONS_MAX_HEIGHT : undefined;
+
+  const onSelectionChange = useCallback(
+    (keys: ListSelection) => {
+      const keyFound = String(head(Array.from(keys)));
+      if (keyFound === SELECT_ALL_OPTION.value) {
+        //@ts-ignore
```

Comment:
`onChange` i feel that this is mistakenly placed here. The `onChange` should be the same as `List`. Therefore we must see over a technical discussion if needed how we can make it happen